gossip/client_test.go

```diff
@@ -138,26 +138,32 @@ func TestClientGossip(t *testing.T) {
 		}
 	}()
 
-	if err := local.AddInfo("local-key", nil, time.Second); err != nil {
+	if err := local.AddInfo("local-key", nil, time.Hour); err != nil {
 		t.Fatal(err)
 	}
-	if err := remote.AddInfo("remote-key", nil, time.Second); err != nil {
+	if err := remote.AddInfo("remote-key", nil, time.Hour); err != nil {
 		t.Fatal(err)
 	}
 
 	// Use an insecure context. We're talking to tcp socket which are not in the certs.
 	rpcContext := rpc.NewContext(&base.Context{Insecure: true}, nil, stopper)
-	client.start(local, disconnected, rpcContext, stopper)
+	disconnected <- client
 
-	util.SucceedsSoon(t, func() error {
```

Comment:
The information lost by panics is anything that has been passed to `t.Error` (or `t.Log`). We've attacked this on two fronts, by avoiding those methods in favor of `t.Fatal` and by using `SucceedsSoon` whenever we need a retry loop. The former is the more important part because not all tests that are susceptible to timeouts are appropriate for a retry loop, but in raft-land where timeouts are a very likely failure mode it's a nice improvement. Some tests also make use of the ability to pass a final error out of the retried function.

Yes, I was suggesting `base.NetworkTimeout` be raised globally.